autoload/ale/code_lens.vim

```diff
@@ -0,0 +1,185 @@
+let s:code_lens_map = {}
+
+" Used to get the code lens map in tests.
+function! ale#code_lens#GetMap() abort
+    return deepcopy(s:code_lens_map)
+endfunction
+
+" Used to set the code lens map in tests.
+function! ale#code_lens#SetMap(map) abort
+    let s:code_lens_map = a:map
+endfunction
+
+function! ale#code_lens#ClearLSPData() abort
+    let s:code_lens_map = {}
+endfunction
+
+" TODO: maybe check with exists instead?
+let s:supported = exists('*nvim_buf_set_virtual_text')
```

Comment:
You have `has('nvim-0.3.2')` in `virtualtext.vim`, why?

Reply:
I can't remember, but I think that was the minimum version requirement due to changes for the functions.